lib/src/json_serializable/type_helper.dart

```diff
@@ -104,7 +105,7 @@ class DateTimeHelper extends TypeHelper {
   bool canDeserialize(DartType type) => _matchesType(type);
 
   bool _matchesType(DartType type) =>
-      type.element.library?.isDartCore == true && type.name == 'DateTime';
```

Comment:
@matanlurey Dude – this is **soooo** much better. I was looking into cleaning this up for a while. Love your solution...